File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.py

class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        ans=0
        maxi=1
        a = [0 for x in range(100)]
        for x in nums: 
            a[x-1]+=1
            if a[x-1]==maxi:
                ans+=1
            if a[x-1]>maxi:
                maxi=a[x-1]
                ans=1
            
        return ans*maxi

File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.cpp
int mp[100];
class Solution {
public:
    
    int maxFrequencyElements(vector<int>& nums) {
        int n=nums.size();
        int ans=0,max=1;
        memset(mp,0,sizeof(mp));

        for( auto it:nums){
            mp[it-1]++;
            if(max==mp[it-1]) ans++;
            if(max<mp[it-1]){
                ans=1;
                max=mp[it-1];
            }
        }
        return ans*max;

    }
};